central/image/service/http_handler_test.go

```diff
@@ -2,29 +2,165 @@ package service
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 
 	"github.com/pkg/errors"
 	"github.com/stackrox/rox/central/imageintegration"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/env"
 	"github.com/stackrox/rox/pkg/features"
 	"github.com/stackrox/rox/pkg/images/enricher"
 	enricherMock "github.com/stackrox/rox/pkg/images/enricher/mocks"
+	"github.com/stackrox/rox/pkg/images/integration/mocks"
+	"github.com/stackrox/rox/pkg/registries/types"
+	scannerMocks "github.com/stackrox/rox/pkg/scanners/mocks"
+	scannerTypes "github.com/stackrox/rox/pkg/scanners/types"
+	scannerTypeMocks "github.com/stackrox/rox/pkg/scanners/types/mocks"
+	scannerv4Mocks "github.com/stackrox/rox/pkg/scannerv4/client/mocks"
 	"github.com/stretchr/testify/assert"
 	"go.uber.org/mock/gomock"
+	"golang.org/x/sync/semaphore"
 )
 
+var _ scannerTypes.Scanner = (*fakeScanner)(nil)
+var _ scannerTypes.SBOMer = (*fakeScanner)(nil)
```

Comment:
Hrm, I'm wondering if there is a a way to avoid having to create these 'mocks' manually.

Reply:
scanner set needs to return a scanner v4 scanner integration to avoid test running into an error since handler verifies scanner v4 integration exsists